salt/modules/timezone.py

```diff
@@ -495,61 +495,66 @@ def set_hwclock(clock):
 
         salt '*' timezone.set_hwclock UTC
     '''
-    os_family = __grains__['os_family']
-    if os_family in ('AIX', 'NILinuxRT'):
-        if clock.lower() != 'utc':
-            raise SaltInvocationError(
-                'UTC is the only permitted value'
-            )
-        return True
-
-    timezone = get_zone()
-
-    if 'Solaris' in __grains__['os_family']:
-        if clock.lower() not in ('localtime', 'utc'):
-            raise SaltInvocationError(
-                'localtime and UTC are the only permitted values'
-            )
-        if 'sparc' in __grains__['cpuarch']:
-            raise SaltInvocationError(
-                'UTC is the only choice for SPARC architecture'
-            )
-        cmd = ['rtc', '-z', 'GMT' if clock.lower() == 'utc' else timezone]
+    if salt.utils.path.which('timedatectl'):
+        cmd = ['timedatectl', 'set-local-rtc',
+               'true' if clock == 'localtime' else 'false']
         return __salt__['cmd.retcode'](cmd, python_shell=False) == 0
+    else:
+        os_family = __grains__['os_family']
```

Comment:
Are we sure these operating systems will hit this block?

Reply:
I was following the style of what was in `get_hwclock` which seemed to be:
```
if systemd_available
    # Use systemd
else
    # Try detect OS
```